### ISSUE TITLE
__pages_cache seems to grow across multiple chapters

### ISSUE THREAD

**Issue by dropwhile** [manga-py/manga-py#390]
# Bug report
Environment (please complete the following information):

Description | Data
---|---
Manga-py version | 1.29.5

#### To Reproduce:
1. Download multiple chapters at once from a site. Arguments used:
```
manga-py \
                -R -W -z -f \
                -d "/some/path" \
                --skip-incomplete-chapters \
                --wait-after-chapter 1 \
                _some_manga_url_with_more_than_a_few_chapters_
```
2. View the ComicInfo.xml files in the last cbz file.  
    `<PageCount>2585</PageCount>` -- should be 20, and in fact _is_ 20 if only that single chapter is downloaded. Since it downloaded many other chapters, it seems the __pages_cache that is used to build this list is held across chapters.
3. Notice that the xml file has far more pages listed than actually exist.

#### Additional context
It seems like [__pages_cache](https://github.com/manga-py/manga-py/blob/1cf0f28a3acba473689f4f6bee75da4f74616664/manga_py/download_methods.py#L44) is not wiped after safe_file is called. download chapters is looped across all chapters in the provider [loop_chapters method](https://github.com/manga-py/manga-py/blob/1cf0f28a3acba473689f4f6bee75da4f74616664/manga_py/provider.py#L176).

It seems resetting the __pages_cache at the start of [downoad_chapter](https://github.com/manga-py/manga-py/blob/1cf0f28a3acba473689f4f6bee75da4f74616664/manga_py/download_methods.py#L46) might be enough to fix the issue.

Example diff:
```
--- download_methods_old.py	2021-04-27 21:26:26.447377237 -0700
+++ download_methods.py	2021-04-27 21:27:05.003609129 -0700
@@ -46,6 +46,9 @@
     def download_chapter(self, chapter_url, path):
         self.chapter_url = chapter_url
         self.path = path
+        # empty __pages_cache for each chapter we download,
+        # so it does not grow across chapters
+        self.__pages_cache = []
         try:
             self.files = self.provider.get_files()
             self._loop_files()
```

**Comment by dropwhile** [comment]
I tested the proposed change (the diff) locally and it seems to behave as expected.

**Comment by 1271** [comment]
Oh. You are right, I forgot to clear pages list
Thanks for the issue. Will be fixed it soon. Thank you!

**Comment by dropwhile** [comment]
fixed by #390. 🎉
